pennylane/compiler/qjit_api.py

```diff
@@ -105,8 +106,177 @@ def circuit(theta):
     """
 
     if not available(compiler):
-        raise RuntimeError(f"The {compiler} package is not installed.")
+        raise CompilerNotFoundError(f"The {compiler} package is not installed.")
 
     compilers = AvailableCompilers.names_entrypoints
     qjit_loader = compilers[compiler]["qjit"].load()
     return qjit_loader(fn=fn, *args, **kwargs)
+
+
+def while_loop(*args, compiler="catalyst", **kwargs):
```

Comment:
Don't forget to add this (and `for_loop`) to the module docstring so that they appear in the docs!

Reply:
Ah, thanks! The automerge is sort of removed this from the module docstring! :(